### PR TITLE
For Code Review

### PR DESCRIPTION
**Pull request by osama-maree**

### CODE REVIEW

File: client-side/package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "react": "^18.2.0",
```
Comment (m5atib) [inline client-side/package.json:13]: Please ignore these files. we do not need to review it

File: client-side/src/component/AlertCustom.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+const AlertCustom = ({ icon, Open, text }) => {
+  return (
+    <Alert
+      severity={icon}
+      action={
+        <IconButton
+          aria-label="close"
+          color="inherit"
+          size="small"
+          onClick={() => {
```
Comment (m5atib) [inline client-side/src/component/AlertCustom.jsx]: Can you please make it shorter like this `onClick={Open}`

Comment (m5atib) [reply client-side/src/component/AlertCustom.jsx]: When the handlers take the same params we can pass it directly

File: client-side/src/App.js [hunk 3]
```diff
@@ -1,9 +1,17 @@
...
+import { Container } from "@mui/material";
+import Home from "./pages/Home.jsx";
 function App() {
   return (
-    <div className="App">
-  
-    </div>
+    <Container
+      maxWidth="sm"
+      sx={{
```
Comment (m5atib) [inline client-side/src/App.js]: Make JSX more cleaner and move this to the CSS file

File: client-side/src/component/Footer.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        marginTop: "20px",
+        marginBottom: "20px",
+        display: "flex",
+        justifyContent: "space-evenly",
+        flexWrap: "wrap",
+      }}
+    >
+      <CustomButton
+        type={"success"}
+        text={`Success Tasks: ${tasks.reduce(
```
Comment (m5atib) [inline client-side/src/component/Footer.jsx]: move these calculations into distributed constants

`const sucessTaks = tasks.reduce((acc, task) => (task.completed ? acc + 1 : acc),0)}`

File: client-side/src/component/Footer.jsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+          (acc, task) => (task.completed ? acc + 1 : acc),
+          0
+        )}`}
+        dis={tasks.length === 0}
+      />
+      <CustomButton
+        type={"warning"}
+        text={`Pending Tasks: ${tasks.reduce(
```
Comment (m5atib) [inline client-side/src/component/Footer.jsx]:
```
const pendingTaks = tasks.reduce(
          (acc, task) => (!task.completed ? acc + 1 : acc),
          0
        )}
```

File: client-side/src/component/Footer.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+          (acc, task) => (!task.completed ? acc + 1 : acc),
+          0
+        )}`}
+        dis={tasks.length === 0}
+      />
+      <CustomButton
+        type={"primary"}
+        text={`ALl Tasks: ${tasks.length}`}
+        dis={tasks.length === 0}
```
Comment (m5atib) [inline client-side/src/component/Footer.jsx]: duplicated usage

can you initiate a constant with a good name like ` const hasTasks = !!tasks.length`

**Review by m5atib — COMMENTED** [review]
please review the needed changes

File: client-side/src/component/Footer.jsx (hunk 6 above)
Comment (osama-maree) [reply client-side/src/component/Footer.jsx]: first usage for number of task
second usage for disabled property

**Pull request merged**